### ISSUE TITLE
JMAP: ContactGroup/set does not handle Contact creationIds correctly

### ISSUE THREAD

**Issue by neilj** [cyrusimap/cyrus-imapd#2451]
When creating/updating a ContactGroup object's `createdIds`, the array passed may include creation ids for Contact objects created in the same API request. Cyrus is not resolving these to the real ids.

**Comment by rsto** [comment]
I guess you mean `contactIds` of the ContactGroup object? So this issue isn't related to the JMAP Contacts API with regards to processing `createdIds`, right?

**Comment by neilj** [comment]
Sorry, yes.

**Comment by rsto** [comment]
Can't reproduce this. Has it been identified already to be a middleware issue?

**Comment by robmueller** [comment]
Yes, this is middleware related.

However I've just added the appropriate `createdIds` object to the JMAP calls, and it appears cyrus is ignoring it and also not returning a `createdIds` response either.

I think we can close this, and I'll open a separate issue about that.

**Comment by robmueller** [comment]
I think this got fixed in the middleware and backend. @neilj can you confirm and close please if yes

**Comment by rsto** [comment]
Which other bugs? AFAIK no-one is currently working on contacts, so we might need to reprioritize the backlog.

**Comment by neilj** [comment]
#2473 and #2474

**Comment by rsto** [comment]
Both issues are fixed now on Cyrus upstream, so you should be OK to retest once this lands on the FM branches.

**Comment by neilj** [comment]
Yep, tested and working. :)
